File: src/test/resources/DeleteTailNode.java
import com.jalgoarena.type.ListNode;

public class Solution {
    /**
     * @param head Linked List head
     * @return  Initial linked list with removed tail
     */
    public ListNode deleteAtTail(ListNode head) {
        if (head == null || head.next == null) {
            return null;
        }

        ListNode preTail = head;

        while (preTail.next.next != null) {
            preTail = preTail.next;
        }

        preTail.next = null;
        return head;
    }
}

File: src/test/resources/DeleteTailNode.kt
import com.jalgoarena.type.ListNode
import com.jalgoarena.type.*

class DeleteTailNode {
    /**
     * @param head Linked List head
     * *
     * @return  Initial linked list with removed tail
     */
    fun deleteAtTail(head: ListNode?): ListNode? {
        if (head == null || head.next == null) {
            return null
        }

        var preTail: ListNode? = head

        while (preTail!!.next!!.next != null) {
            preTail = preTail.next
        }

        preTail.next = null
        return head
    }
}
